docs/data/sdks/next-gen-sdks.md

```diff
@@ -0,0 +1,186 @@
+# flush queued events every 20 seconds
+client.configuration.flush_interval_millis = 20000
+```
+
+### Storage provider
+
+A storage provider holds events in a designated storage buffer. The default storage provider only queues events in memory. To prevent data lose, you can customize your own storage provider to persist events, for example to a disk, in case of unexpected shutdown. This way, you can ensure that you can recover un-sent events even in the event of a system failure. 
+
+You can configure the storage provider in the Configuration of your Client. Refer to the "Configuration Options" section of a specific next-gen SDK full documentation for more details.
+
+### Logger provider
+
+A logger provider configure the logger instance used by Amplitude Client. It helps to collect debugging and error messages from the SDK in both development and production environments.
+
+You can configure the logger provider in the Configuration of your Client. Refer to the "Configuration Options" section of a specific next-gen SDK full documentation for more details.
+
+### Event 
+
+Events represent how users interact with your application. 
+
+`BaseEvent` represents a basic event with optional properties. You can also track other event properties in the field of `EventProperties`
+
+```python
+# Next-gen Python SDK example: Event
+
+from amplitude import Amplitude, BaseEvent
+
+client = Amplitude('API_KEY')
+
+# Track a basic event
+# One of user_id and device_id is required
+event = BaseEvent(event_type="Button Clicked", user_id="USER_ID")
+client.track(event)
+
+# Track events with optional properties
+client.track(
+    BaseEvent(
+        event_type="Button Clicked",
+        user_id="USER_ID",
+        app_version="1.2.1" # optional, the current version of your application
+        event_properties={ # other event properties you want to track 
+            "color": "red"
+        }
+))
+```
+
+### Plugin
+
+Plugins allow you to extend Amplitude SDK's behavior by, for example:
+
+- Use an Enrichment Plugin to modify event properties
+- Use a Destination Plugin to send events to a third-party APIs
+
+A plugin is an object with methods `setup()` and `execute()`:
+
+- `setup()` method contains logic for preparing the plugin for use and has Client instance as a parameter. The expected return value is None. A typical use for this method is to copy configuration from `client.configuration` or instantiate plugin dependencies. This method is called when you register the plugin to the client via `client.add()`.
+
+- `execute()` method the logic for processing events and has event instance as parameter. The expected return value for an Enrichment Plugin is the modified/enriched event, while the expected return value for a Destination Plugin is a map with keys: event (BaseEvent), code (HTTP response status code), and message (string). This method is called for each event that's instrumented using the client interface.
```

Comment:
We should call out that if `execute()` returns no event, the event will not propagate through the remaining plugins. It could be good to also include an example of a "drop events" plugin which is also a common ask from customers.